### PR TITLE
fix(core): do not readConfigs in progress of provision

### PR DESCRIPTION
**Pull request by jayzhang**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/9948007

### CODE REVIEW

File: packages/fx-core/src/core/index.ts [hunk 1]
```diff
@@ -798,7 +798,7 @@ class CoreImpl implements Core {
 
   private async setSubscription(subscription: SubscriptionInfo | undefined) {
     if (subscription) {
-      await this.readConfigs();
+      // await this.readConfigs();
```
Comment (jayzhang) [inline packages/fx-core/src/core/index.ts:801]: do not readConfigs() in the progress of provision

**Review by LongOddCode — APPROVED** [review]

File: packages/fx-core/src/core/index.ts [hunk 2]
```diff
@@ -885,30 +893,44 @@ class CoreImpl implements Core {
...
-        const solutionConfig = entry[1];
-        const configJson = mapToJson(solutionConfig);
-        const filePath = `${this.ctx.root}/.${ConfigFolderName}/env.${envName}.json`;
-        const localDataPath = `${this.ctx.root}/.${ConfigFolderName}/${envName}.userdata`;
-        const localData = sperateSecretData(configJson);
-        const content = JSON.stringify(configJson, null, 4);
-        await fs.writeFile(filePath, content);
-        await fs.writeFile(localDataPath, serializeDict(localData));
+    let res: Result<null, FxError> = ok(null);
+    for (let i = 0; i < 5; ++i) {
```
Comment (1yefuwang1) [inline packages/fx-core/src/core/index.ts:904]: Why are we doing this loop 5 times.

File: packages/fx-core/src/core/index.ts [hunk 3]
```diff
@@ -885,30 +893,44 @@ class CoreImpl implements Core {
...
+          this.ctx.logProvider?.debug(
+            `[Core] writeConfigs()#########file:${jsonFilePath}, content:${content}`
+          );
+        }
+        //write settings
+        await this.writeSettings(this.ctx.root, this.ctx.projectSettings);
+        break;
+      } catch (e) {
+        res = err(error.WriteFileError(e));
+        sleep(10);
```
Comment (1yefuwang1) [inline packages/fx-core/src/core/index.ts:930]: Why do we need to sleep here.

File: packages/fx-core/src/core/index.ts (hunk 2 above)
Comment (jayzhang) [reply packages/fx-core/src/core/index.ts:904]: It's guarantee to make sure config is written successfully

File: packages/fx-core/src/core/index.ts (hunk 3 above)
Comment (jayzhang) [reply packages/fx-core/src/core/index.ts:930]: To avoid concurrent write.

Comment (1yefuwang1) [reply packages/fx-core/src/core/index.ts]: Besides，I think await is needed here

**Review by 1yefuwang1 — APPROVED** [review]

**Review by MSFT-yiz — APPROVED** [review]

**Pull request merged**